georocket-common/src/main/java/io/georocket/query/DefaultQueryCompiler.java

```diff
@@ -262,16 +265,31 @@ public void enterString(StringContext ctx) {
         }
       }
     }
-    
+
     @Override
-    public void enterKeyvalue(KeyvalueContext ctx) {
+    public void enterLt(LtContext ctx) {
       currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.LT.name());
     }
-    
+
+    @Override
+    public void enterGt(GtContext ctx) {
+      currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.GT.name());
+    }
+
+    @Override
+    public void enterEq(EqContext ctx) {
+      currentKeyvalue = new JsonObject();
+      currentKeyvalue.put("comp", Comparator.EQ.name());
+    }
+
     @Override
     public void exitKeyvalue(KeyvalueContext ctx) {
```

Comment:
Discussion: Are there any drawbacks in ignoring the "exitEq", "exitGt", "exitLt" events but using "exitKeyvalue" instead for all cases.